dev/AppLifecycle/AppLifecycle.cpp

```diff
@@ -38,7 +49,33 @@ namespace winrt::Microsoft::ProjectReunion::implementation
         return {commandLine.substr(argsStart, argsLength), commandLine.substr(dataStart)};
     }
 
-    Windows::ApplicationModel::Activation::IActivatedEventArgs AppLifecycle::GetActivatedEventArgs()
+    bool IsEncodedLaunch(IProtocolActivatedEventArgs const& args)
+    {
+        return (args.Uri().SchemeName() == c_encodedLaunchSchemeName);
+    }
+
+    IActivatedEventArgs GetEncodedLaunchActivatedEventArgs(IProtocolActivatedEventArgs const& args)
+    {
+        for (auto const& pair : args.Uri().QueryParsed())
+        {
+            if (pair.Name() == L"ContractId")
```

Comment:
>ContractId [](start = 33, length = 10)

Is this case-sensitive? Is "contractid" equally valid?

The RFC for URIs says it's a per-protocol decision
[https://stackoverflow.com/questions/24699643/are-query-string-keys-case-sensitive](https://stackoverflow.com/questions/24699643/are-query-string-keys-case-sensitive)

Reply:
Nope, and I have had plans on changing this and one or two other places.  It's just going to be more resilient if we ignore case here.  The initial writing of this needs touching up.